internal/services/purview/purview_account_resource.go

```diff
@@ -43,55 +42,135 @@ func resourcePurviewAccount() *pluginsdk.Resource {
+				Default:  true,
+			},
+
+			"managed_resource_group_name": {
+				Type:         pluginsdk.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ForceNew:     true,
+				ValidateFunc: resourcegroups.ValidateName,
+			},
+
+			"identity": commonschema.SystemOrUserAssignedIdentityRequired(),
+
+			"managed_resources": {
+				Type:     pluginsdk.TypeList,
+				Computed: true,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"resource_group_id": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+						"storage_account_id": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+						"event_hub_namespace_id": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"catalog_endpoint": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
+
+			"guardian_endpoint": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
+
+			"scan_endpoint": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
+
+			"atlas_kafka_endpoint_primary_connection_string": {
+				Type:      pluginsdk.TypeString,
+				Computed:  true,
+				Sensitive: true,
+			},
+
+			"atlas_kafka_endpoint_secondary_connection_string": {
+				Type:      pluginsdk.TypeString,
+				Computed:  true,
+				Sensitive: true,
+			},
+
+			"tags": commonschema.Tags(),
+		},
 	}
 }
 
-func resourcePurviewAccountCreateUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+func resourcePurviewAccountCreate(d *pluginsdk.ResourceData, meta interface{}) error {
 	client := meta.(*clients.Client).Purview.AccountsClient
 	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
-	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
 	id := account.NewAccountID(subscriptionId, d.Get("resource_group_name").(string), d.Get("name").(string))
-	if d.IsNewResource() {
-		existing, err := client.Get(ctx, id)
-		if err != nil {
-			if !response.WasNotFound(existing.HttpResponse) {
-				return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
-			}
-		}
 
+	existing, err := client.Get(ctx, id)
+	if err != nil {
 		if !response.WasNotFound(existing.HttpResponse) {
-			return tf.ImportAsExistsError("azurerm_purview_account", id.ID())
+			return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
 		}
 	}
 
+	if !response.WasNotFound(existing.HttpResponse) {
+		return tf.ImportAsExistsError("azurerm_purview_account", id.ID())
+	}
+
 	purviewAccount := account.Account{
 		Properties: &account.AccountProperties{},
-		Location:   utils.String(azure.NormalizeLocation(d.Get("location").(string))),
-		Tags:       tags.Expand(d.Get("tags").(map[string]interface{})),
+		Location:   pointer.To(location.Normalize(d.Get("location").(string))),
+	}
+
+	if v, ok := d.GetOk("tags"); ok {
```

Comment:
Is there a reason you pulled this out into a `d.GetOk`? It does work this way but the other way it was written is the way we do it across the bulk of our resources

Reply:
I'll put it back, good point. My only reason was my habit using GetOk for optional properties.